portal-impl/src/com/liferay/portal/service/impl/UserServiceImpl.java

```diff
@@ -3528,6 +3528,9 @@ protected long[] checkOrganizations(long userId, long[] organizationIds)
 
 		PermissionChecker permissionChecker = getPermissionChecker();
 
+		boolean strictAssignment = PropsValues.ORGANIZATIONS_ASSIGNMENT_STRICT ||
+								   !permissionChecker.isCompanyAdmin();
+
```

Comment:
can we just inline this? so we don't need to think a name of this variable 😄

Reply:
no, I get it to avoid checking the permission in the loop.
Can you move it before the loop where you're using the variable?